aten/src/ATen/TensorOptions.h

```diff
@@ -17,10 +17,29 @@ namespace at {
 
 /// A class to encapsulate construction axes of a `Tensor`.
 /// `TensorOptions` is a virtual class to enable overriding of certain methods
-/// by subclasses in other libraries, such as PyTorch. In PyTorch, there is a
-/// `torch::TensorOptions` subclass of this `TensorOptions`, which changes
-/// `type()` to return a variable type instead of a tensor type, such that
-/// variables are created inside factory methods, instead of tensors.
+/// by subclasses in other libraries, such as PyTorch.
+///
+/// WARNING: In PyTorch, there are torch:: variants of factory functions,
```

Comment:
This file isn't really about the tensor factories (like `at::ones`), so the comment feels a bit out of place to me, but I guess more documentation doesn't hurt

Reply:
"The information you need, even when you don't expect it."